renderdoc/driver/gl/gl_manager.cpp

```diff
@@ -28,9 +28,13 @@
 #include "driver/gl/gl_driver.h"
 
 GLResourceManager::GLResourceManager(WrappedOpenGL *driver)
-    : ResourceManager(), m_Driver(driver), m_SyncName(1)
+    : m_State(driver->GetState()), ResourceManager(m_State), m_Driver(driver), m_SyncName(1)
```

Comment:
I don't understand how this is intended to work. This will get the state value at the time the resource manager is created, but then it won't update because `driver->GetState()` doesn't return a reference. It's shared with the parent `ResourceManager` by reference but that doesn't help if `GLResourceManager::m_State` isn't updated. I don't see where it would get updated after that.

What I was suggesting is that you change the resource manager to be e.g. `GLResourceManager::GLResourceManager(WrappedOpenGL *driver, CaptureState &driverState)` and then pass the reference through to `ResourceManager`'s constructor as you do here. The `m_State` in `GLResourceManager` can be removed since there's no need to have it duplicated from the parent class. That means when the driver creates it, it can pass in a reference to `m_State` then so it will be kept up to date.

For a similar example look at the constructors of `WrappedID3D12CommandQueue` and `WrappedID3D12GraphicsCommandList` which does that already to share the state with the parent device without needing updating.

Reply:
Oh, so we want a direct reference from the driver _itself_, not from a specific manager. OK.

(it currently works only tracking by reference `m_State` of a specific manager. So at least when the specific manager changes its `m_State`, the `m_DriverState` in the base manager gets its value)

To reiterate I get it correctly now:

- We update `driver->GetState()` to return a reference
- We remove local `m_State` from all specific managers, and move it as a reference to base manager
- We initialize the base manager with the passed reference directly from the driver.